docs.md

```diff
@@ -0,0 +1,1366 @@
+
+Files uploaded to this account will be **publicly visible**. Be very careful not to upload anything with sensitive information like passwords or private data.
+
+</div>
+```
+
+<div class = "warning">
+<b style="color: rgb(var(--color-highlight));">Warning!</b><br>
+
+Files uploaded to this account will be **publicly visible**. Be very careful not to upload anything with sensitive information like passwords or private data.
+
+</div>
+
+## Including non-interactive code
+
+Include inline code with single backticks: `library(ggplot2)`. 
+
+Make code blocks with at least three backticks:
+
+````
+```r
+# You only need to install it once
+install.packages("ggplot2")
+
+# You'll need to load the library anew for each R session
+library("ggplot2")
+```
+````
+
+```r
+# You only need to install it once
+install.packages("ggplot2")
+
+# You'll need to load the library anew for each R session
+library("ggplot2")
+```
+
+You don't have to specify the programming language, but you can, and it should help you get appropriate syntax highlighting.
+
+````
+```python
+print("This is python code")
+```
+````
+
+```python
+print("This is python code")
+```
+
+## Including interactive code
+
+We use three main approaches for interactive coding exercises: 
+
+- We can link to interactive notebooks hosted on binder (see example notebooks in [our R notebook repository](https://github.com/arcus/education_r_environment)). This is our approach for most R modules, for example: [R Basics: Introduction](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/main/r_basics_introduction/r_basics_introduction.md) 
+- We can have interactive coding cells in modules themselves using sagemath. This is our approach for most python modules, for example: [Python Basics: Writing Code](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/main/python_basics_writing_python_code/python_basics_writing_python_code.md)
+- We can also have interactive coding cells in modules using AlaSQL, which is our approach for SQL modules, for example: [SQL Basics](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/main/sql_basics/sql_basics.md)
+
+We don't have a way to include interactive code for git or bash. For those modules, we just instruct learners to work on their own machines.
+
+### Interactive R
+
+If your lesson includes R code examples hosted in binder from [our R notebook repository](https://github.com/arcus/education_r_environment), then you'll need to load the R macros by adding the following `import` statement to the front matter of your module:
+
+```
+import: https://raw.githubusercontent.com/arcus/education_modules/templates_update/_module_templates/macros_r.md
+```
+
+You can then insert the `@lesson_prep_r` macro, which includes instructions for opening up the relevant rmd document, in your Lesson Preparation section:
+
```

Comment:
Additional front matter guidelines are being created, so we should make sure to update this section to include things like `coding_language`, `depends_on_knowledge_available_in`, etc. and make clear what is optional and when various designations are required.

Reply:
Yes, absolutely! There will be new front matter items coming from the wrapper template and macros as well (and possibly the exercise ones?). What do you think about getting this merged as the base version of the documentation, and then having updates to docs.md be included in all the other various PRs that would add new front matter fields? Rather than trying to make this version of docs.md complete before those other files are quite ready yet.

Reply:
I agree, let's merge what you already made, and then add to that.